collection/collection.go

```diff
@@ -0,0 +1,78 @@
+package collection
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qri-io/qri/collection/list"
+	"github.com/qri-io/qri/event"
+	"github.com/qri-io/qri/profile"
+)
+
+// RootUserKeyID is a special key identifer that lists *all* datasets in a
+// collection
+const RootUserKeyID = profile.ID("root")
+
+type Collection interface {
+	List(ctx context.Context, pid profile.ID, lp list.Params) ([]Item, error)
+}
+
+type EditableCollection interface {
+	Collection
+	Put(ctx context.Context, items ...Item) error
+	Delete(ctx context.Context, profileID profile.ID, initIDs ...string) error
+}
+
+// Item is a dataset in a collection of datasets
+// it's a slightly-modified version of the flatbuffer defined in dscache.fbs
+type Item struct {
```

Comment:
On the flip-hand, I do feel a bit itchy with this one, from a store perspective, this is a lot of fields and a lot of those fields are duplicated / de-normalized. I have to look into how this would be generated from what we have at hand (ie `dataset.Dataset` or `VersionInfo`) and have it be combined with some `WorkflowInfo` which has yet to be added.

Reply:
agreed it requires a lot more thought. I've added clarifying comments to both the `Collection` interface and the `Item` struct def. I'm hoping they help justify what's owned by the collection and what is cached.

I think we still have a long way to go comparing `VersionInfo` to `Item`